pkg/controller/hyperconverged/hyperconverged_controller.go

```diff
@@ -416,6 +416,68 @@ func (r *ReconcileHyperConverged) Reconcile(request reconcile.Request) (reconcil
 		// One shortcoming is that only one failure of a particular condition can be
 		// captured at one time (ie. if KubeVirt and CDI are both reporting !Available,
 		// you will only see CDI as it updates last).
+
+		if cond, ok := r.conditions[conditionsv1.ConditionDegraded]; ok {
```

Comment:
honestly I'm a bit confused about this one:
r.conditions in the in memory copy of the conditions we are going to write on the hyperconverged resource.
We are going to write them only on line 481.

In precedence we updated r.conditions scanning all the components (but what's going to happen if we don't finish or if we skip one?).
Now we are overriding it.
I'm a bit scared we can loose something.